projects/igniteui-angular/src/lib/grids/grid-navigation.service.ts

```diff
@@ -301,6 +301,7 @@ export class IgxGridNavigationService {
     }
 
     public shouldPerformHorizontalScroll(visibleColIndex: number, rowIndex = -1) {
+        if (visibleColIndex < 0 || visibleColIndex > this.grid.columnList.length - 1) { return false; }
```

Comment:
Since the columnList contains all columns (including hidden columns), shouldn't this check for the visibleColumns instead?